### PR TITLE
Fix the broken code-gen on mixer/adapter/configs.

### PR DESCRIPTION
**Pull request by guptasu**
Also changed the script to curl files into a temp instead of vendor, which was causing dirty vendor upon `go generate`.

### CODE REVIEW

File: bin/mixer_codegen.sh [hunk 1]
```diff
@@ -99,27 +99,41 @@ fi
...
 
-mkdir -p ${ROOT}/vendor/github.com/googleapis/googleapis
+mkdir -p ${tmpdir}/googleapis/googleapis/google/rpc
 
 # all the google_rpc protos
-mkdir -p ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc
-curl -sS ${GOOGLEAPIS_URL}/google/rpc/status.proto > ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc/status.proto
-curl -sS ${GOOGLEAPIS_URL}/google/rpc/code.proto > ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc/code.proto
-curl -sS ${GOOGLEAPIS_URL}/google/rpc/error_details.proto > ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc/error_details.proto
+curl -sS ${GOOGLEAPIS_URL}/google/rpc/status.proto > ${tmpdir}/googleapis/googleapis/google/rpc/status.proto
```
Comment (ldemailly) [inline bin/mixer_codegen.sh]: can't we get those from vendor instead ?

File: bin/mixer_codegen.sh [hunk 2]
```diff
@@ -99,27 +99,41 @@ fi
 
 # TODO: vendor those too like istio.io/api without pruning protos
 
+tmpdir=$(dirname $(mktemp -u))/protos_for_mixer_codegen
+
 GOOGLEAPIS_SHA=c8c975543a134177cc41b64cbbf10b88fe66aa1d
 GOOGLEAPIS_URL=https://raw.githubusercontent.com/googleapis/googleapis/${GOOGLEAPIS_SHA}
 
-if [ ! -e ${ROOT}/vendor/github.com/googleapis/googleapis ]; then
```
Comment (ldemailly) [inline bin/mixer_codegen.sh]: should this work ?

**Comment by kyessenov** [comment]
Yes, they live here now:
https://github.com/gogo/googleapis/tree/master/google/rpc

**Comment by ldemailly** [comment]
there is already https://github.com/istio/vendor-istio/tree/master/github.com/googleapis
why cant it have everything we need ?
we should not be curling anything

**Comment by ldemailly** [comment]
we have this too:
https://github.com/istio/vendor-istio/tree/master/github.com/gogo/googleapis/google

so it's just a matter of fixing import paths in the code and we can delete all the curl badness?

File: bin/mixer_codegen.sh [hunk 3]
```diff
@@ -99,27 +99,41 @@ fi
...
 fi
 
 imports=(
  "${ROOT}"
  "${ROOT}/vendor/istio.io/api"
- "${ROOT}/vendor/github.com/gogo/protobuf"
- "${ROOT}/vendor/github.com/gogo/protobuf/protobuf"
- "${ROOT}/vendor/github.com/googleapis/googleapis"
+ "${tmpdir}/gogo/protobuf"
+ "${tmpdir}/googleapis/googleapis"
```
Comment (ldemailly) [inline bin/mixer_codegen.sh]: let's add what is missing ? (assuming it's a few files, which it is if I read the curl right)

**Comment by guptasu** [comment]
thanks @kyessenov and @ldemailly . Didn't knew about the prune magic. Sending another PR to use the pruning stuff.

**Pull request closed without merging**